### DISCUSSION TITLE
Pagination Primitive Component Requirement

### DISCUSSION THREAD

**Discussion by zhaohuanyuu** [system-ui/theme-ui#2345] [Q&A] [answered]
Are future releases considering adding pagination to the primitivecomponent?It is also used very often ：）

**Comment by lachlanjc** [comment, marked as answer]
Not currently planned, no! Theme UI Components isn't a full-service UI kit; we're providing some shortcuts for using your theme styles with the most common HTML elements, plus showing how to extend it for your own design system. With something like a pagination component, because it's composed many different elements, there's more variations in the design than we want to support, & each of the elements can use lower-level Theme UI components like Link or Button.

That said, the `sidenav` package which we use on the docs site does have a [publicly-available Pagination component](https://github.com/system-ui/theme-ui/blob/develop/packages/sidenav/src/index.tsx#L375), but it's deeply intertwined with the MDX sidebar design, if you want to check that out.

**Reply by zhaohuanyuu** [reply]
OK, then use my own. I just make sure to avoid repeating in the future
